pkg/labels/labels.go

```diff
@@ -201,6 +201,23 @@ func (ls Labels) Has(name string) bool {
 	return false
 }
 
+// WithoutEmpty returns the labelset without empty labels.
+// May return the same labelset.
+func (ls Labels) WithoutEmpty() Labels {
+	for _, v := range ls {
+		if v.Value == "" {
```

Comment:
minor: 
```
if v.Value != "" {
   continue
}
```
.. would make a complex path the "main one" in this function (aka filter boring cases first) (: